src/main/java/com/amazon/corretto/hotpatch/HotPatchMain.java

```diff
@@ -0,0 +1,148 @@
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.corretto.hotpatch;import java.io.BufferedReader;
+import java.io.File;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.util.List;
+import java.util.Properties;
+import java.util.Set;
+
+import com.sun.tools.attach.VirtualMachine;
+import sun.jvmstat.monitor.MonitoredHost;
+import sun.jvmstat.monitor.MonitoredVm;
+import sun.jvmstat.monitor.MonitoredVmUtil;
+import sun.jvmstat.monitor.VmIdentifier;
+
+import static com.amazon.corretto.hotpatch.Constants.LOG4J_FIXER_AGENT_VERSION;
+import static com.amazon.corretto.hotpatch.Logger.log;
+
+public class HotPatchMain {
+    public static void main(String[] args) throws Exception {
+        Logger.setVerbose(args);
+
+        String pid[];
+        if (args.length == 0) {
+            MonitoredHost host = MonitoredHost.getMonitoredHost((String)null);
+            Set<Integer> pids = host.activeVms();
+            pid = new String[pids.size()];
+            int count = 0;
+            for (Integer p : pids) {
+                MonitoredVm jvm = host.getMonitoredVm(new VmIdentifier(p.toString()));
+                String mainClass = MonitoredVmUtil.mainClass(jvm, true);
+                if (!myName.equals(mainClass)) {
+                    log(p + ": " + mainClass);
+                    pid[count++] = p.toString();
+                }
+            }
+            if (false && count > 0) {
+                log("Patch all JVMs? (y/N) : ");
+                BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
+                String answer = in.readLine();
+                if (!"y".equals(answer)) {
+                    System.exit(1);
+                    return;
+                }
+            } else if (count > 0) {
+                log("Patching all JVMs!");
+            }
+        } else if (args.length == 1 && ("-h".equals(args[0]) || "-help".equals(args[0]) || "--help".equals(args[0]))) {
+            System.out.println("usage: Log4jHotPatch [<pid> [<pid> ..]]");
+            System.exit(1);
+            return;
+        } else {
+            pid = args;
+        }
+        boolean succeeded = loadInstrumentationAgent(pid);
+        if (succeeded) {
+            System.exit(0);
+        } else {
+            log("Errors occurred deploying hot patch. If you are using java 8 to run this\n" +
+                    "tool against JVM 11 or later, the target JVM may still be patched. Please look for a message\n" +
+                    "like 'Loading Java Agent (using ASM 6).' in stdout of the target JVM. Also note that JVM 17+\n" +
+                    "are not supported.");
+            System.exit(1);
+        }
+    }
+
+    private static String myName = HotPatchAgent.class.getName();
```

Comment:
Move to top of file

Reply:
This was being used to avoid patching ourselves. But now that we have three different entry points, we need to ensure we don't match any of them, so it is replaced with a check against a list.